src/views/MoviePage.vue

```diff
@@ -1,13 +1,151 @@
 <script>
 import { defineComponent } from 'vue';
 
+import { getMovieDetails } from '../api/monteCinemaApi';
+
+import BaseTag from '@/components/global/BaseTag.vue';
+
 export default defineComponent({
-  name: 'MoviePage'
+  name: 'MoviePage',
+  components: {
+    BaseTag
+  },
+  data() {
+    return {
+      id: this.$route.params.movieId,
```

Comment:
I had to really fight this one out :sweat_smile:
Turns out that the watcher doesn't work if the `immediate: true` option comes after the logic...
Well, now it works and I'm not going to ever forget this.

Reply:
> Turns out that the watcher doesn't work if the immediate: true option comes after the logic...

wut? 🤨
that's definitely not true: https://codesandbox.io/s/keen-ritchie-i6g49v?file=/src/components/HelloWorld.vue

maybe there was something wrong with nesting e.g.
```js
watch: {
  movieId: {
    handler(movieId) {
      this.movie = await getMovieDetails(movieId);
    },
  }
  immediate: true,
}
```
instead of 

```js
watch: {
  movieId: {
    handler(movieId) {
      this.movie = await getMovieDetails(movieId);
    },
    immediate: true,
  }
}
```